CMakeLists.txt

```diff
@@ -3,7 +3,13 @@ project(poco_vendor VERSION "1.1.1")
 
 # Can work with poco 1.4.1p1 (earliest to use recursive mutexes on Linux)
 # 1.6.1 is the first version to ship with PocoConfigVersion.cmake
-find_package(Poco "1.6.1" COMPONENTS Foundation QUIET)
+if(NOT ROS2_FORCE_BUILD_ALL_VENDOR)
+  find_package(Poco "1.6.1" COMPONENTS Foundation QUIET)
```

Comment:
Did you try setting `-DPOCO_DIR=...` to force the use of the right Poco?

https://cmake.org/cmake/help/v3.0/command/find_package.html

```
Search paths specified in cmake-specific environment variables. These are intended to be set in the user’s shell configuration. This can be skipped if NO_CMAKE_ENVIRONMENT_PATH is passed:

<package>_DIR
```

Reply:
Well, the `ExternalProject` logic exists here for a reason. All I want here is to disable calling `find_package(Poco` in favor of doing the logic block below it.